common/components/inventory.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from typing import Optional, TYPE_CHECKING
+
+from common.engine import component
+from common.utils.item.inventory_item import InventoryItem
+from common.utils.vector2 import Vector2
+
+if TYPE_CHECKING:
+    from typing import Callable
+
+
+@component
+class Inventory:
+    __items: list[Optional[InventoryItem]]
+    __size: Vector2
+
+    def __init__(self, size: Vector2):
```

Comment:
`component` is `dataclass` so you should use `__post_init__` instead of `__init__`